code/game/machinery/doppler_array.dm

```diff
@@ -50,8 +52,10 @@ GLOBAL_LIST_EMPTY(doppler_arrays)
 												  took,orig_dev_range,orig_heavy_range,orig_light_range)
 	if(stat & NOPOWER)
 		return FALSE
+	if(next_announce > world.time)
+		return
+	next_announce = world.time + cooldown

```

Comment:
Shouldn't this only get a cooldown for explosions within range? I imagine it may be annoying that explosions on another z level would prevent your from detecting explosions soon after within range

Reply:
Oh yeah, not sure why i put it above that check.